### PR TITLE
Sensor card update to copy fill feature

### PR DESCRIPTION
**Pull request by zsarnett**
## Proposed change
This PR updates the Core Sensor card to mimic the fill feature of the Custom Graph Card

image

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml
detail: 1
entity: sensor.outside_humidity
graph: line
type: sensor
```

## Additional information
PR also moves the icon to the right side. Updates some CSS and CSS classes

- Link to documentation pull request: Soon

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by zsarnett** [comment]
image

File: src/panels/lovelace/cards/hui-sensor-card.ts [hunk 1]
```diff
@@ -175,6 +192,7 @@ class HuiSensorCard extends LitElement implements LovelaceCard {
       detail: 1,
       theme: "default",
       hours_to_show: 24,
+      fill: true,
```
Comment (balloob) [inline src/panels/lovelace/cards/hui-sensor-card.ts]: Should we make this configurable or should we just always set it to `true` ?

Comment (zsarnett) [reply src/panels/lovelace/cards/hui-sensor-card.ts]: I'm down to make it always true. I think it looks way better like this

Comment (balloob) [reply src/panels/lovelace/cards/hui-sensor-card.ts]: Alright. Let's drop it .

File: src/panels/lovelace/cards/hui-sensor-card.ts [hunk 2]
```diff
@@ -69,34 +80,40 @@ const calcPoints = (
...
+      last = [average(item), lastValue(item)];
+    }
+    const y = height - ((item ? last[0] : last[1]) - min) / yRatio;
     return coords.push([x, y]);
   };
 
-  history.forEach((item, i) => getCoords(item, i, 0, detail));
+  for (let i = 0; i < history.length; i += 1) {
+    getCoords(history[i], i, 0, detail);
+  }
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-sensor-card.ts]: Why this change?

Comment (zsarnett) [reply src/panels/lovelace/cards/hui-sensor-card.ts]: This is for when there isn't enough history to fill the hours requested. The previous method would start the line in the middle of the card if there were only 12 hours of data when 24 were requested

Comment (bramkragten) [reply src/panels/lovelace/cards/hui-sensor-card.ts]: Might be me, but I don't see a difference in what it does, just a more difficult way to write the same ;-)

Comment (zsarnett) [reply src/panels/lovelace/cards/hui-sensor-card.ts]: image

The difference using the previous method. For loop starts at 0 and loops through no matter if the item is []. For Each ignores NULL/Undefined/[] items

Comment (zsarnett) [reply src/panels/lovelace/cards/hui-sensor-card.ts]: Therefore i starts at ex. 14. Meaning the first X value is 14 * xRatio

**Comment by balloob** [comment]
Please don't force push after reviews have started, it prevents us from spotting the changes.

**Review by balloob — APPROVED** [review]

**Comment by balloob** [comment]
🎉 🐬 🎉 🐬 🎉 🐬
